vmhooks/bigIntOps.md

```diff
@@ -250,124 +257,128 @@ module BIGINTOPS
          </instrs>
          <locals> 0 |-> <i32> _DST  1 |-> <i32> OP1_IDX  2 |-> <i32> OP2_IDX </locals>
          <bigIntHeap> HEAP </bigIntHeap>
-      requires notBool (#validIntId(OP1_IDX, HEAP))
-        orBool notBool (#validIntId(OP2_IDX, HEAP))
+      requires notBool (OP1_IDX in_keys{{HEAP}})
+        orBool notBool (OP2_IDX in_keys{{HEAP}})
 
     // extern void bigIntSub(void* context, int32_t destination, int32_t op1, int32_t op2);
     rule <instrs> hostCall("env", "bigIntSub", [ i32 i32 i32 .ValTypes ] -> [ .ValTypes ]) => . ... </instrs>
          <locals> 0 |-> <i32> DST  1 |-> <i32> OP1_IDX  2 |-> <i32> OP2_IDX </locals>
-         <bigIntHeap> HEAP => HEAP [DST <- {HEAP[OP1_IDX]}:>Int -Int {HEAP[OP2_IDX]}:>Int] </bigIntHeap>
-      requires #validIntId(OP1_IDX, HEAP)
-       andBool #validIntId(OP2_IDX, HEAP)
+         <bigIntHeap> HEAP
+                   => HEAP {{ DST <- (HEAP{{OP1_IDX}} orDefault 0) -Int (HEAP{{OP2_IDX}} orDefault 0) }}
+         </bigIntHeap>
+      requires OP1_IDX in_keys{{HEAP}}
+       andBool OP2_IDX in_keys{{HEAP}}
       [preserves-definedness]
       // Preserving definedness:
-      //  - {HEAP[OP*_IDX]}:>Int is defined because #validIntId(OP*_IDX, HEAP)
       //  - -Int is total
-      //  - Map[Kitem <- KItem] is total
+      //  - in_keys is total
+      //  - _{{_ <- _}} is total
```

Comment:
You should also include: `_{_} orDefault _` is total.

The same for the similar changes in this file.

BTW, I wonder if this rule still needs the "preserves-definedness" attribute... It might, because of the `<locals>` cell, but the booster backend may be smart enough to handle that properly.

Reply:
I updated  the comments, thanks. 
I agree, the backend should handle them without the attributes. I'll try removing `preserves-definedness` once we receive the `in_keys` update.